pandas/tests/frame/test_duplicates.py

```diff
@@ -477,3 +477,46 @@ def test_drop_duplicates_inplace():
     expected = orig2.drop_duplicates(["A", "B"], keep=False)
     result = df2
     tm.assert_frame_equal(result, expected)
+
+
+@pytest.mark.parametrize(
+    "origin_dict, output_dict, ignore_index, output_index",
+    [
+        ({"A": [2, 2, 3]}, {"A": [2, 3]}, True, [0, 1]),
+        ({"A": [2, 2, 3]}, {"A": [2, 3]}, False, [0, 2]),
+        ({"A": [2, 2, 3], "B": [2, 2, 4]}, {"A": [2, 3], "B": [2, 4]}, True, [0, 1]),
+        ({"A": [2, 2, 3], "B": [2, 2, 4]}, {"A": [2, 3], "B": [2, 4]}, False, [0, 2]),
+    ],
+)
+def test_drop_duplicates_ignore_index_inplace_false(
+    origin_dict, output_dict, ignore_index, output_index
+):
+    # GH 30114
+    df = DataFrame(origin_dict)
+    result = df.drop_duplicates(ignore_index=ignore_index)
+
+    expected = DataFrame(output_dict, index=output_index)
+    tm.assert_frame_equal(result, expected)
+
+    # to verify if original dataframe is not mutated
+    tm.assert_frame_equal(df, DataFrame(origin_dict))
+
+
+@pytest.mark.parametrize(
```

Comment:
can you incorporate these in the above test as well.

can you also assert if inplace== True that the input is unchanged (copy it before and check equality)

Reply:
incorporated, and added test

thanks @jreback